torch/csrc/autograd/custom_function.cpp

```diff
@@ -215,6 +215,13 @@ optional_variable_list _process_backward_mode_ad(
                          bool is_differentiable) {
     if (!is_differentiable) {
       if (!var.requires_grad()) {
+        if (is_input) {
+          // We need this in case forward AD runs, since forward AD assumes that the output
+          // processed by backward ad can never be an input as-is.
+          // If forward AD does not run, we do some extra work here, but that is OK
+          AutoGradMode grad_mode(false);
```

Comment:
- Can we have this in a separate function that is called here and below?
- Didn't we add a context to disable forward mode AD while doing this view in a previous PR??

Reply:
- sounds good
- that was the initial plan in the previous PR, but we eventually decided not to in favor of just returning early in the _process_forward_ad function. (we did disable forward AD while calling user-defined jvp)

Reply:
Hmm maybe I am missing something? My understanding is that we had two options:
1. if forward is no-op, we require user to manually return a view in jvp (otherwise error) to match the behavior of what backward does (when the output does require grad)
2. if forward is no-op, automatically return a view in jvp (we did this by continuing to run the view with forward grad enabled, and just adding a `continue` as we loop through the outputs in _process_forward_ad, as to avoid setting the forward grad again).

We ended up going with (2), but we didn't consider the case in _process_backward_ad when the output is considered non-differentiable (perhaps because it does not require grad), so in this PR, we add performing the view in this case as well.